### PR TITLE
fix panic when restart after removeMember

### PR DESCRIPTION
**Pull request by yangxuanjia**
I used the ETCD3.5 version to stably reproduce this problem, and the process is divided into the following steps:
1. Establish a 3 node cluster A, B, C
2. Write a few keys casually
3. Make an automatic snapshot for each node
use this code quickly do an auto snapshot:  https://github.com/etcd-io/etcd/pull/13474
4. removeMember a node C
5. Restart node A, -----> Panic

https://github.com/etcd-io/etcd/issues/13466

The logic of the overall error is basically clear. The key is to take a snapshot before deleting the node.
Then the biggest problem is in the above code. If the index of the snapshot is less than the ConsistentIndex, start with the old backend. The old backend has the latest member information (including the deleted node information), but the WAL playback is based on the snapshotIndex. Play back at the starting position, so the removeMember raft changeConfig will be played back.

### CODE REVIEW

File: server/etcdserver/api/rafthttp/transport.go [hunk 1]
```diff
@@ -343,8 +343,9 @@ func (t *Transport) removePeer(id types.ID) {
 		peer.stop()
 	} else {
 		if t.Logger != nil {
-			t.Logger.Panic("unexpected removal of unknown remote peer", zap.String("remote-peer-id", id.String()))
+			t.Logger.Warn("skipped removing already removed remote peer", zap.String("remote-peer-id", id.String()))
```
Comment (ahrtr) [inline server/etcdserver/api/rafthttp/transport.go:346]: It's strange to see

d

Comment (yangxuanjia) [reply server/etcdserver/api/rafthttp/transport.go]: see again

**Comment by serathius** [comment]
PTAL @ptabor

**Comment by ahrtr** [comment]
> PTAL @ptabor

@serathius  @ptabor  This issue can't be reproduced on 3.5.1, so no need to get it included in the plans for v3.5.2. Please see my comments [issuecomment-973757868](https://github.com/etcd-io/etcd/issues/13466#issuecomment-973757868) and [issuecomment-975024314](https://github.com/etcd-io/etcd/issues/13466#issuecomment-975024314). 

But I agree that it is indeed an issue in the main branch (3.6).

**Comment by ahrtr** [comment]
Please also squash the two commits and rebase this PR.

**Comment by yangxuanjia** [comment]
Three submits are inserted with a lot of Push, Squash failed, and re-raised a PR to integrate commit.
https://github.com/etcd-io/etcd/pull/13645

**Pull request closed without merging**